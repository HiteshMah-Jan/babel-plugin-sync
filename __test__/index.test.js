const babel = require('@babel/core')
const sync = require('../')

function snapshotTest({ title, code, debug }) {
  test(title, () => {
    const res = babel.transform(code, {
      plugins: [sync],
      babelrc: false
    })
    if (debug) {
      console.log(res.code)
    } else {
      expect(res.code).toMatchSnapshot()
    }
  })
}

snapshotTest({
  title: 'main',
  code: `
  class Foo {
    // $MakeMeSync
    async bar() {
      await bar()
      await this.bar()
    }
  }`
})

snapshotTest({
  title: 'manually',
  code: `
  class Foo {
    // $MakeMeSync
    /**
     * jsdoc
     *
     */
    async bar() {
      return this.bar() // $MakeMeSync
    }

    // trailing
  }
  `
})
