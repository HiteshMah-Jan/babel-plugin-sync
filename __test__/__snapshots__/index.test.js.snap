// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"class Foo {
  // __SYNCED__
  async bar() {
    await bar();
    await this.bar();
  }

  barSync() {
    barSync();
    this.barSync();
  }

}"
`;

exports[`manually 1`] = `
"class Foo {
  // __SYNCED__

  /**
   * jsdoc
   *
   */
  async bar() {
    return this.bar(); // $MakeMeSync
  }

  barSync() {
    return this.barSync();
  } // trailing


}"
`;
